core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -151,34 +151,34 @@ open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
       url?.substringAfterLast("/", "")
         ?.takeIf { it.contains(".") }
         ?: src?.substringAfterLast("/", "")
-          ?.substringAfterLast("%3A", "") ?: ""
+          ?.substringAfterLast("%3A") ?: ""
 
     @SuppressWarnings("NestedBlockDepth")
     override fun handleMessage(msg: Message) {
       val url = msg.data["url"] as? String
       val src = msg.data["src"] as? String
       if (url != null || src != null) {
         val fileName = getDecodedFileName(url, src)
-        val dotIndex = fileName.lastIndexOf('.')
         var root =
           Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES)
         if (instance.externalMediaDirs.isNotEmpty()) {
           root = instance.externalMediaDirs[0]
         }
-        var storageDir = File(root, fileName)
-        var newUrl = fileName
-        var i = 2
-        while (storageDir.exists()) {
-          newUrl = fileName.substring(0, dotIndex) + "_" + i + fileName.substring(dotIndex)
-          storageDir = File(root, newUrl)
-          i++
-        }
+        val fileToSave = sequence {
+          yield(File(root, fileName))
+          yieldAll(generateSequence(1) { it + 1 }.map {
+            File(
```

Comment:
ctrl+shift+j until this file declaration is all 1 line

Reply:
done